Editor/Scripts/CodeGenerators/RGCodeGenerator.cs

```diff
@@ -577,22 +627,71 @@ private static void PopulateObjectTypes(List<RGStatesInfo> statesInfos, List<RGA
                 }
             }
             
-            // TODO (REG-1373) iterate through all scenes rather than only the current one in the editor
-            // For objects in the scene
-            RGEntity[] allEntities = Object.FindObjectsOfType<RGEntity>();
-            for (int i = 0; i < allEntities.Length; i++)
+            // iterate through all scenes rather than only the current ones in the editor
+            var startingActiveScenePath = SceneManager.GetActiveScene().path;
+            List<string> allActiveScenePaths = new ();
+            HashSet<string> allLoadedScenePaths = new ();
+            for (int j = 0; j < SceneManager.sceneCount; j++)
             {
-                RGEntity entity = allEntities[i];
-                var entityMap = entity.MapObjectType(objectTypeMap);
-                foreach (var kvp in entityMap)
+                var scene = SceneManager.GetSceneAt(j);
+                allActiveScenePaths.Add(scene.path);
+                if (scene.isLoaded)
                 {
-                    if (kvp.Value != null)
-                    {
-                        objectTypeMap[kvp.Key] = kvp.Value;
-                    }
+                    allLoadedScenePaths.Add(scene.path);
                 }
             }
             
+            //sort the activeScenePaths so that the unloaded ones are at the end
+            // this matters later when we reload them
+            allActiveScenePaths.Sort((a, b) =>
+            {
+                if (allLoadedScenePaths.Contains(a))
+                {
+                    return -1;
+                }
+                return allLoadedScenePaths.Contains(b) ? 1 : 0;
+            });
+            
+            // get all the objects in the currently open scenes.. this minimizes the amount of scene loading we have to do
+            LookupEntitiesForCurrentScenes(objectTypeMap);
+            
+            // Get for all the other scenes in the build 
+            EditorBuildSettingsScene[] scenesInBuild = EditorBuildSettings.scenes;
+            foreach (var editorScene in scenesInBuild)
```

Comment:
If we're only searching scenes that have been added toe the build settings, we're introducing an edge case where scenes exist in the project but have not been added to build settings yet. The order of `Create Scene`->`Generate Scripts`->`Add scene to build settings` wouldn't work. Just to be 100% sure we're not missing scenes I would either 
* regenerate every time a scene is added to build settings
* search the actual directory for `.unity` files and get paths that way
* search directory for `.unity` files and compare the names to what `EditorBuildSettings.scene` and show a warning of "this scene won't be included in generated data" etc

Reply:
If their scene isn't in their build settings, I wouldn't want us to be interrogating it for data as it won't be part of their runtime anyway and there may be a very good reason it is excluded; it may not even be in a working state.

I have added a sentence to the warning prompt to clarify what is included.

![image](https://github.com/Regression-Games/RGUnityBots/assets/112959031/f301b611-6b1e-468e-817b-95882bd39ad5)